src/traffic participants/SpecialTransport.ts

```diff
@@ -0,0 +1,27 @@
+class SpecialTransport implements RoadUser{
+    velocity: number;
+    direction: Direction;
+    isRoadFree: boolean;
+    priority: Priority;
+
+    constructor(vel: number, dir: Direction) {
```

Comment:
You use the same constructor 3 times. It would be better to create a fathering class that will contain this structure and then call its constructor with just "super(vel, dir)"